openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +116,261 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
                 (ungraded_problem_2.location, {'due': None}),
                 (graded_problem_1.location, {'due': 5}),
             ]
-            sequence = modulestore().get_item(sequence.location)
             self.assertCountEqual(_gather_graded_items(sequence, 5), expected_graded_items)
+
+    def test_get_custom_pacing_children(self):
+        """
+        _get_custom_pacing_items should return a list of (block item location, field metadata dictionary)
+        where the due dates are set from due_num_weeks
+        """
+        # A subsection with multiple units but no problems
+        with modulestore().bulk_operations(self.course.id):
+            sequence = ItemFactory(parent=self.course, category='sequential', due_num_weeks=2)
+            vertical1 = ItemFactory(parent=sequence, category='vertical')
+            vertical2 = ItemFactory(parent=sequence, category='vertical')
+            vertical3 = ItemFactory(parent=sequence, category='vertical')
+            expected_dates = [
+                (sequence.location, {'due': timedelta(weeks=2)}),
+                (vertical1.location, {'due': timedelta(weeks=2)}),
+                (vertical2.location, {'due': timedelta(weeks=2)}),
+                (vertical3.location, {'due': timedelta(weeks=2)})
+            ]
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+        # A subsection with multiple units, each of which has a problem
+            problem1 = ItemFactory(parent=vertical1, category='problem')
+            problem2 = ItemFactory(parent=vertical1, category='problem')
+            expected_dates.extend([(problem1.location, {'due': timedelta(weeks=2)}),
+                            (problem2.location, {'due': timedelta(weeks=2)})])
+            sequence = modulestore().get_item(sequence.location)
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+        # A subsection that has ORA as a problem
+            ItemFactory.create(parent=vertical3, category='openassessment')
+            sequence = modulestore().get_item(sequence.location)
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+        # A subsection that has an ORA problem and a non ORA problem
+            problem4 = ItemFactory(parent=vertical3, category='problem')
+            expected_dates.append((problem4.location, {'due': timedelta(weeks=2)}))
+            sequence = modulestore().get_item(sequence.location)
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+
+@ddt.ddt
+class SelfPacedCustomDueDateTests(SharedModuleStoreTestCase):
```

Comment:
Let's talk about if this needs to be its own test class (I'm honestly not sure, but would be good to talk through)
I also think we should look into how we can separate out these into individual tests. I think for readability, it will help to have some of the course creation happening inside of a test rather than all at the top.